apitools/base/py/credentials_lib.py

```diff
@@ -37,6 +36,21 @@
 from apitools.base.py import exceptions
 from apitools.base.py import util
 
+# App Engine does not support ctypes which are required for the
+# monotonic time used in fasteners. Conversely, App Engine does
+# not support colocated concurrent processes, so process locks
+# are not needed.
+try:
+    import fasteners
+    _ENV_IS_APP_ENGINE = False
```

Comment:
I think you could also try inspecting `os.environ['SERVER_SOFTWARE']`, but I'm not sure it's worth it.

WDYT of just making this `_FASTENERS_AVAILABLE` and then ignoring locking otherwise? Anyone who's `pip install`'d will have it anyway.

Reply:
Sounds good.